src/main/java/org/jenkinsci/plugin/gitea/GiteaDeleteSCMEvent.java

```diff
@@ -0,0 +1,177 @@
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String description() {
+        String ref = getPayload().getRef();
+        ref = ref.startsWith(Constants.R_HEADS) ? ref.substring(Constants.R_HEADS.length()) : ref;
+        ref = ref.startsWith(Constants.R_TAGS) ? ref.substring(Constants.R_TAGS.length()) : ref;
+        String refType = getPayload().getRefType();
+        return "Delete event for " + refType + " " + ref + " in repository " +
+                getPayload().getRepository().getOwner().getUsername() + "/" +
+                getPayload().getRepository().getName();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @NonNull
+    @Override
+    public Map<SCMHead, SCMRevision> headsFor(GiteaSCMSource source) {
+        String ref = getPayload().getRef();
+        ref = ref.startsWith(Constants.R_HEADS) ? ref.substring(Constants.R_HEADS.length()) : ref;
+        ref = ref.startsWith(Constants.R_TAGS) ? ref.substring(Constants.R_TAGS.length()) : ref;
+        String refType = getPayload().getRefType();
+        if ("branch".equals(refType)) {
+            BranchSCMHead h = new BranchSCMHead(ref);
+            return Collections.<SCMHead, SCMRevision>singletonMap(h, null); //new BranchSCMRevision(h, null));
+        } else if ("tag".equals(refType)) {
+            TagSCMHead h = new TagSCMHead(ref, System.currentTimeMillis());
+            return Collections.<SCMHead, SCMRevision>singletonMap(h, null); //new TagSCMRevision(h, null));
+        } else {
+            return Collections.<SCMHead, SCMRevision>emptyMap();
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean isMatch(@NonNull SCM scm) {
+        URIish uri;
+        try {
+            uri = new URIish(getPayload().getRepository().getHtmlUrl());
+        } catch (URISyntaxException e) {
+            return false;
+        }
+        String ref = getPayload().getRef();
+        ref = ref.startsWith(Constants.R_HEADS) ? ref.substring(Constants.R_HEADS.length()) : ref;
+        if (scm instanceof GitSCM) {
+            GitSCM git = (GitSCM) scm;
+            if (git.getExtensions().get(IgnoreNotifyCommit.class) != null) {
+                return false;
+            }
+            for (RemoteConfig repository : git.getRepositories()) {
+                for (URIish remoteURL : repository.getURIs()) {
+                    if (GitStatus.looselyMatches(uri, remoteURL)) {
+                        for (BranchSpec branchSpec : git.getBranches()) {
+                            if (branchSpec.getName().contains("$")) {
+                                // If the branchspec is parametrized, always run the polling
+                                return true;
+                            } else {
+                                if (branchSpec.matches(repository.getName() + "/" + ref)) {
+                                    return true;
+                                }
+                            }
```

Comment:
To improve readability this could be changed into (probably comment needs to be updated to cover both cases):
```suggestion
                            if (branchSpec.getName().contains("$") || branchSpec.matches(repository.getName() + "/" + ref)) {
                                // If the branchspec is parametrized, always run the polling
                                return true;
                            }
```

Reply:
Like I mentioned in my latest PR comment I am not sure if we even need this method implemented. It is just copied from the corresponding CreateEvent impl. I have not been able to get this part of the code triggered runtime so I don't even know which scenario it is about to cover...